OneSignalSDK/onesignal/core/src/test/java/com/onesignal/user/internal/subscriptions/SubscriptionManagerTests.kt

```diff
@@ -18,11 +20,52 @@ import io.mockk.mockk
 import io.mockk.runs
 import io.mockk.spyk
 import io.mockk.verify
+import junit.framework.TestCase
 import org.junit.runner.RunWith
 
 @RunWith(KotestTestRunner::class)
 class SubscriptionManagerTests : FunSpec({
 
+    test("Deadlock related to Model.setOptAnyProperty") {
+        // Given
+        val modelStore = MockHelper.configModelStore()
+        val model = modelStore.model
+
+        val t1 =
+            Thread {
+                // acquire "model.data", then trigger the onChanged event
+                model.setOptAnyProperty("key1", "value1")
+            }
+
+        val t2 =
+            Thread {
+                // acquire "model.initializationLock", then wait for "model.data" to be released
+                model.initializeFromModel("", MockHelper.configModelStore().model)
+            }
+
+        model.subscribe(
+            object : IModelChangedHandler {
+                // will be executed in t1
+                override fun onChanged(
+                    args: ModelChangedArgs,
+                    tag: String,
+                ) {
+                    Thread.sleep(200)
+                    // waiting for "model.initializationLock"
+                    model.toJSON()
+                }
+            },
+        )
+
+        t1.start()
+        t2.start()
+
+        // Give some time for the thread to complete the task
+        Thread.sleep(1000)
```

Comment:
The new commit replaces Thread.sleep with t2.join(1000) where 1000 is the timeout time of 1 second. If I don't add the timeout, the test unit will wait indefinitely because the threads can never finish in the deadlock scenario.